### DISCUSSION TITLE
Shouldn't `side-effect-cache` set to false by default in CI environments?

### DISCUSSION THREAD

**Discussion by ayZagen** [pnpm/pnpm#7993] [Ideas]
I spent hours to find why bcrypt fails in container built from CI but not from my local machine. TIL (correct me if i am wrong pls.), pnpm doesn't execute postinstall scripts for cached dependencies. Even `pnpm rebuild bcrypt` was using cached ones.

Setting `side-effect-cache=false` in `.npmrc` resolved the issue. But i think pnpm should behave like this by default in CI environments. 

There could be several things would result in a faulty containers, such as using distributed caches with a different arch/os, multi-layered builds etc. In this situation the error was discovered in runtime, meaning build stages passed successfully despite of it was faulty.

What do you think?

@zkochan I didn't open an issue for this but I really would like to hear your thoughts about it.

**Comment by doseofted** [comment]
I've also ran into this issue across a few different projects using pnpm (typically when the `@go-task/cli` package is installed) and had trouble finding a solution. Today I'll typically set the same option `sideEffectsCache: false` in my pnpm-workspace configuration as a workaround. This error seems to happen whenever the pnpm store is present. It will also sometimes happen on my development machine if I clear out the node modules folder and reinstall with a pnpm store on the hard drive.

I think a more clear notice in the documentation about the issue would be helpful for others (or a change in the defaults while this issue persists).
